drivers/serial/uart_rpi_pico.c

```diff
@@ -74,12 +76,22 @@ static int uart_rpi_init(const struct device *dev)
 	 * uart_init() may be replaced by register based API once rpi-pico platform
 	 * has a clock controller driver
 	 */
-	baudrate = uart_init(uart_inst, data->baudrate);
+	baudrate = uart_init(uart_inst, data->uart_config.baudrate);
 	/* Check if baudrate adjustment returned by 'uart_init' function is a positive value */
-	if (baudrate <= 0) {
+	if (baudrate == 0) {
 		return -EINVAL;
 	}
-
+	/*
+	 * initialize uart_config with hardware reset values
+	 * https://datasheets.raspberrypi.com/rp2040/rp2040-datasheet.pdf#tab-registerlist_uart page:431
+	 */
+	data->uart_config = (struct uart_config){
+		.baudrate = baudrate,
+		.data_bits = UART_CFG_DATA_BITS_8,
+		.flow_ctrl = UART_CFG_FLOW_CTRL_NONE,
+		.parity = UART_CFG_PARITY_NONE,
+		.stop_bits = UART_CFG_STOP_BITS_1
+	};
```

Comment:
Do you need to call `uart_set_format()` here, since the hardware defaults to 5-bit at reset? Also might be a good idea to document that the data bit setting is different than hardware reset.

Reply:
Added an extra comment the deviation from hw default. 
Added a call for `uart_set_format()`